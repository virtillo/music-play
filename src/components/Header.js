import React from 'react';

// Header Component. It will include the search input when the prop showSearch is passed.
const Header = (props) => {

    let search;
    if (props.showSearch){
        search = (
            <input 
                onChange={props.handleSearchTermChange} 
                value={props.searchTerm} 
                type="text" 
                placeholder="Search" 
                className="search"
            /> 
        );
    }
    return(
        <header>
            <article className="header content-wrapper">
                <section className="logo">
                    <img src="../img/logo.png" alt="MusicPlay" />
                </section>
                {search}
            </article>
        </header>
    );
};

export default Header;