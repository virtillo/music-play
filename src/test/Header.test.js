import React from 'react';
import renderer from 'react-test-renderer';
import Header from '../components/Header';


test('Header renders correctly', () => {
    const component = renderer.create(<Header/>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});