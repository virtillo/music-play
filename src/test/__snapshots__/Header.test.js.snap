// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header>
  <article
    className="header content-wrapper"
  >
    <section
      className="logo"
    >
      <img
        alt="MusicPlay"
        src="../img/logo.png"
      />
    </section>
  </article>
</header>
`;

exports[`Home renders correctly 1`] = `
<header>
  <article
    className="header content-wrapper"
  >
    <section
      className="logo"
    >
      <img
        alt="MusicPlay"
        src="../img/logo.png"
      />
    </section>
  </article>
</header>
`;
